lager/lenses.hpp

```diff
@@ -123,51 +129,169 @@ auto attr(Member member)
     });
 }
 
+/*!
+ * `Key -> Lens<{X}, [X]>`
+ */
 template <typename Key>
-auto at(Key key)
-{
-    return zug::comp([=](auto&& f) {
-        return [f, &key](auto&& p) {
-            return f([&] {
+auto at(Key key) {
```

Comment:
I like this new version, but I think the old behavior is more consistent with what C++ often does, for example with `operator[]`. I would add a new version called `at_opt` for this.

Reply:
I disagree with this on several points:
- [`vector::operator[]`](https://en.cppreference.com/w/cpp/container/vector/operator_at) does _not_ return a default constructed element when the index is out of bounds. accessing an OOB index is, in fact undefined behavior.
- [`map::operator[]`](https://en.cppreference.com/w/cpp/container/map/operator_at) _does_ return a default constructed element, _but also adds it to the collection_. your implementation explicitly forbids writing at an OOB index.
- the name of this lens is `at(idx)`, which, both in the `std` api and any other api, is able to signal the abscence of a value (`std::out_of_bounds`, `Maybe`, etc).

Maybe having a lens named `at_or(idx, [value])` could be the solution here?